content/docs/19-kubernetes-knowlege-hub/00-how-to/02-how-to-retrieve-images-from-private-registry.md

```diff
@@ -0,0 +1,185 @@
+
+To log into a Docker registry from Kubernetes, you must create a secret containing your registry credentials. You can use this secret in a Kubernetes deployment configuration to pull images from the registry.
+
+In this tutorial, you will log into a private docker registry to pull existing images of an application you will deploy in Kubernetes.
+
+## Prerequisites
+
+- The kubectl [command-line tool](https://kubernetes.io/docs/tasks/tools/install-kubectl-linux/). Kubectl allows you to connect to, configure and work with your clusters through the command line.
+- Access to a private registry.  [DockerHub](https://hub.docker.com/) offers a single private registry on the free tier. If you do not have a personal registry account, you can use DockerHub.
+- Access to a running Kubernetes cluster. Setting up a Kubernetes cluster can be difficult. You can read about how to [create one from scratch](https://www.notion.so/How-To-Create-a-Kubernetes-Cluster-bf707518b6bf4a918d8b11a570eabed6) or on [Spectro Cloud’s console.](https://docs.spectrocloud.com/getting-started/#deployingyourfirstcluster)
+
+Below is an example of how you can create a secret and use it in a Kubernetes deployment:
+
+## **Create a JSON File Containing your Credentials.**
+
+First, create a file called **registry-creds.json** that contains your registry credentials in the following format:
+
+```json
+{
+	"auths": {
+		"example.registry.com": {
+			"username": "username",
+			"password": "password"
+		}
+	}
+}
+```
+
+Keeping passwords in plain text is unsafe. Kubernetes automatically encodes passwords used to create a secret in base64.  Encoding passwords does not mean your passwords cannot be decoded. 
+
+## **Create a Kubernetes Secret from the JSON File Containing the Secrets.**
+
+Use the kubectl command-line tool to generate a secret from the **registry-creds.json** file:
+
+```bash
+kubectl create secret generic myregistrykey --from-file=registry-creds.json
+```
+
+You can use the command below to view the secret created in detail.
+
+```bash
+kubectl get secret/myregistrykey -o json
+```
+
+The output of the command displays that the content of the **registry-creds.json** file is base 64 encoded.
+
+```json
+{
+    "apiVersion": "v1",
+    "data": {
+        "registry-creds.json": "ewogICJhdXRocyI6IHsKICAgICJleGFtcGxlLnJlZ2lzdHJ5LmNvbSI6IHsKICAgICAgInVzZXJuYW1lIjogInRlc3RfdXNlcm5hbWUiLAogICAgICAicGFzc3dvcmQiOiAidGVzdF9wYXNzd29yZCIKICAgIH0KICB9Cn0K"
+    },
+    "kind": "Secret",
+    "metadata": {
+        "creationTimestamp": "2023-03-22T08:44:26Z",
+        "name": "myregistrykey",
+        "namespace": "default",
+        "resourceVersion": "1552285",
+        "uid": "ccfb047b-67c8-446b-a69a-6eb762c3100f"
+    },
+    "type": "Opaque"
+}
+```
+
+You will decode the secret you created to verify that secrets are not secure.
```

Comment:
I'm not sure what's happening here. Is this a "test" of sorts to decode the secret? And the test is to confirm that secrets are not secure? "To verify that secrets are not secure" is unclear. Maybe it's something like "To show you that...?" 

I'm not sure of the point being made here.

Suggestion: "Invoke the following command to ..."